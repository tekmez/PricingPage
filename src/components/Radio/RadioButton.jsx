import React from 'react'
import { View, TouchableWithoutFeedback } from 'react-native'
import styles from './Radio.style'
import CheckIcon from '../../../assets/svg/checkIcon.svg'
const RadioButton = ({ isEnabled, setIsEnabled }) => {
  const onSelect = () => {
    setIsEnabled((previousState) => !previousState)
  }
  return (
    <TouchableWithoutFeedback testID="radio-button" onPress={onSelect}>
      <View testID="radio-container" style={styles.radioButtonContainer}>
        <View
          testID="radio-button-style"
          style={
            isEnabled
              ? styles.radioButtonSelected
              : styles.radioButtonUnselected
          }
        >
          {isEnabled && <CheckIcon width={14} height={14} />}
        </View>
      </View>
    </TouchableWithoutFeedback>
  )
}

export default RadioButton
