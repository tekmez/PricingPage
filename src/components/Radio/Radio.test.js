import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import RadioButton from './RadioButton'
import styles from './Radio.style'

jest.mock('../../../assets/svg/checkIcon.svg', () => () => (
  <div testID="check-icon" />
))

describe('Radio', () => {
  it('renders custom radio component', () => {
    const comp = render(<RadioButton />).toJSON()
    expect(comp).toMatchSnapshot()
  })

  it('renders with initial state', () => {
    const { getByTestId } = render(<RadioButton isEnabled={false} />)
    const radioButton = getByTestId('radio-button')
    expect(radioButton).toBeTruthy()
  })

  it('toggles isEnabled state when pressed', () => {
    const setIsEnabledMock = jest.fn()
    const { getByTestId } = render(
      <RadioButton isEnabled={false} setIsEnabled={setIsEnabledMock} />
    )
    const radioButton = getByTestId('radio-button')
    fireEvent.press(radioButton)
    expect(setIsEnabledMock).toHaveBeenCalledWith(expect.any(Function))
  })

  it('renders with color for false state', () => {
    const setIsEnabledMock = jest.fn()
    const { getByTestId } = render(
      <RadioButton isEnabled={false} setIsEnabled={setIsEnabledMock} />
    )
    const radioButtonStyle = getByTestId('radio-button-style').props.style
    expect(radioButtonStyle).toMatchObject(styles.radioButtonUnselected)
  })

  it('renders with color for true state', () => {
    const setIsEnabledMock = jest.fn()
    const { getByTestId } = render(
      <RadioButton isEnabled={true} setIsEnabled={setIsEnabledMock} />
    )
    const radioButtonStyle = getByTestId('radio-button-style').props.style
    expect(radioButtonStyle).toMatchObject(styles.radioButtonSelected)
  })
})
