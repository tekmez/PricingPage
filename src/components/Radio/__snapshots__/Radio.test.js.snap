// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders custom radio component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
  testID="radio-button"
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#E7E8E9",
        "borderRadius": 12,
        "borderWidth": 2,
        "height": 24,
        "justifyContent": "center",
        "width": 24,
      }
    }
    testID="radio-button-style"
  />
</View>
`;
